charts/hivemq-platform-operator/templates/deployment.yml

```diff
@@ -156,12 +156,25 @@ spec:
             successThreshold: 1
             timeoutSeconds: 10
           resources:
+            {{- if hasKey .Values.resources "overrideLimits" }}
+            {{- with .Values.resources.overrideLimits }}
             limits:
-              cpu: {{.Values.resources.cpu | quote}}
-              memory: {{.Values.resources.memory | quote}}
+              {{- toYaml . | nindent 14 }}
```

Comment:
Do we need to quote the yaml snippet, or is this even possible? 

If I configure: 
 resources:
    cpu: 1024m
    memory: 2048M
    overrideLimits:
      cpu: 2048m
      memory: 4096M

I get: 
             resources:
                limits:
                  cpu: 2048m
                  memory: 4096M
                requests:
                  cpu: "1024m"
                  memory: "2048M"

Reply:
There is no need to quote the YAML snippet, usually it's a good practice to quote those values which are string format (cpu, memory, etc), to avoid potential issues with others such as float, integer or boolean. But in this particular case it will converted to a string by default even if you just only type digits.

Even if you ran a `kubectl apply` for a manifest with the following:

```yaml
limits:
  cpu: 2048m
  memory: 4096M
requests:
  cpu: "1024m"
  memory: "2048M"
```

You would only see everything without any quotes in the Kubernetes API
<img width="332" alt="image" src="https://github.com/user-attachments/assets/4fc2e1f1-8f86-4b4a-bacd-e07ba52aa73d">

Reply:
As I mentioned, for this particular case, it's not really needed, so I'm going to remove the quote function from the template, to get these values aligned.
I tested as well with a running platform to make sure there is no rolling restart out of it because of this change.